### PR TITLE
Add servicerating base

### PR DESCRIPTION
**Pull request by imsickofmaps**
Custom API format is: https://gist.github.com/imsickofmaps/066568878b9ebbf7a985

### CODE REVIEW

**Comment by imsickofmaps** [comment]
@smn ready for review

File: servicerating/api.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        # Get the pre-existing User Account, we don't except everything
+        user_account = UserAccount.objects.get(key=bundle.obj.user_account)
+        # Get the pre-existing Conversation, we don't except everything
+        conversation = Conversation.objects.get(key=bundle.obj.conversation_key)
+        contact = Contact(conversation=conversation,
+                          key = bundle.obj.contact["key"],
+                          value = bundle.obj.contact,
+                          msisdn = bundle.obj.contact["msisdn"])
+        contact.save()
+        # print contact.pk
```
Comment (smn) [inline servicerating/api.py]: removable

**Comment by smn** [comment]
:+1: but slightly out of my depth with the custom resource stuff with tastypie, have you got a link for me to read up on that?
Also, watch your db indexes for that `extra` & `contact` stuff, that's likely going to turn slow as things go big.

**Comment by imsickofmaps** [comment]
Thanks: I modelled the custom resource stuff on the example from a Riak TastyPie example: http://django-tastypie.readthedocs.org/en/latest/non_orm_data_sources.html

**Comment by imsickofmaps** [comment]
And on the slowness thing, I was already thinking about that. I'm planning on adding a custom reporting API for Sapphire to consume in front of it that we can optimise by pre-calculating if it gets slow.

**Pull request merged**